e2e/test/scenarios/question/offset.cy.spec.ts

```diff
@@ -1,43 +1,395 @@
+import { uuid } from "metabase/lib/uuid";
+import type {
+  Aggregation,
+  Breakout,
+  FieldReference,
+  StructuredQuery,
+} from "metabase-types/api";
 
-const { ORDERS_ID } = SAMPLE_DATABASE;
+const { ORDERS, ORDERS_ID, PRODUCTS } = SAMPLE_DATABASE;
+
+const ORDERS_TOTAL_FIELD_REF: FieldReference = [
+  "field",
+  ORDERS.TOTAL,
+  { "base-type": "type/Float" },
+];
+
+const ORDERS_CREATED_AT_BREAKOUT: Breakout = [
+  "field",
+  ORDERS.CREATED_AT,
+  { "base-type": "type/DateTime", "temporal-unit": "month" },
+];
+
+const PRODUCTS_CATEGORY_BREAKOUT: Breakout = [
+  "field",
+  PRODUCTS.CATEGORY,
+  { "base-type": "type/text", "source-field": ORDERS.PRODUCT_ID },
+];
+
+const SUM_TOTAL_AGGREGATION: Aggregation = ["sum", ORDERS_TOTAL_FIELD_REF];
+
+const OFFSET_SUM_TOTAL_AGGREGATION_NAME = "Offsetted sum of total";
+
+const OFFSET_SUM_TOTAL_AGGREGATION: Aggregation = [
+  "offset",
+  createOffsetOptions(OFFSET_SUM_TOTAL_AGGREGATION_NAME),
+  SUM_TOTAL_AGGREGATION,
+  -1,
+];
 
 describe("scenarios > question > offset", () => {
   beforeEach(() => {
     restore();
     cy.signInAsAdmin();
+    cy.intercept("POST", "/api/card").as("saveQuestion");
+  });
+
+  it("does not work without a breakout", () => {
+    const query: StructuredQuery = {
+      "source-table": ORDERS_ID,
+      aggregation: [OFFSET_SUM_TOTAL_AGGREGATION],
+    };
+
+    createQuestion({ query }, { visitQuestion: true });
+
+    verifyQuestionError(
+      "Window function requires either breakouts or order by in the query",
+    );
+  });
+
+  it("works with a single breakout", () => {
+    const query: StructuredQuery = {
+      "source-table": ORDERS_ID,
+      aggregation: [OFFSET_SUM_TOTAL_AGGREGATION],
+      breakout: [ORDERS_CREATED_AT_BREAKOUT],
+      limit: 5,
+    };
+
+    createQuestion({ query }, { visitQuestion: true });
+
+    verifyNoQuestionError();
+    verifyTableContent([
+      ["April 2022", ""],
+      ["May 2022", "52.76"],
+    ]);
+  });
+
+  it("works with a single breakout and sorting by breakout", () => {
+    const query: StructuredQuery = {
+      "source-table": ORDERS_ID,
+      aggregation: [OFFSET_SUM_TOTAL_AGGREGATION],
+      breakout: [ORDERS_CREATED_AT_BREAKOUT],
+      limit: 5,
+      "order-by": [["desc", ORDERS_CREATED_AT_BREAKOUT]],
+    };
+
+    createQuestion({ query }, { visitQuestion: true });
+
+    verifyNoQuestionError();
+    verifyTableContent([
+      ["April 2026", "45,683.68"],
+      ["March 2026", "47,403.97"],
+    ]);
+  });
+
+  it.skip("works with a single breakout and sorting by aggregation (metabase#42554)", () => {
+    const query: StructuredQuery = {
+      "source-table": ORDERS_ID,
+      aggregation: [OFFSET_SUM_TOTAL_AGGREGATION],
+      breakout: [ORDERS_CREATED_AT_BREAKOUT],
+      limit: 5,
+      "order-by": [["desc", ["aggregation", 0]]],
+    };
+
+    createQuestion({ query }, { visitQuestion: true });
+
+    verifyNoQuestionError();
+    /* TODO: assert actual values */
```

Comment:
are you going to merge it like this? or is it a leftover?

Reply:
Merge like this for now.

This test is a repro for not-yet-fixed bug and is skipped for now.
I added info about updating the assertion in the relevant issue: #42554